### DISCUSSION TITLE
Scroll lock areas of the screen

### DISCUSSION THREAD

**Discussion by rigrig** [scandum/tintin#229] [Ideas]
I'd like to draw buttons "on top" of the buffer, something like this (inspired by BlowTorch):

![image](https://github.com/user-attachments/assets/a538cd2d-ab06-46e2-8a2e-0d113c393cde)

Would it be doable to add something like `#buffer {lock} {top left bottom right}` (and `#buffer {unlock} {top left bottom right}`), making the buffer skip printing into those areas?

**Comment by scandum** [comment]
It's not really feasible, as it'd be quite a bit of work.

The easiest solution would be to redraw when data is received from the MUD.

```
#event {RECEIVED OUTPUT} {#buffer refresh;#delay 0 #draw Red box 10 -20 15 -1}
```
It might also be an option to use #split to create a slide bar when tapping the right side of the screen. It's also possible to make the vt automap clickable, and #help MSLP describes how to create links, which could be used on exits for movements, mobs to attack them, objects to pick them up, etc.

**Reply by rigrig** [reply]
Thanks, I was afraid you'd say that.

Redrawing it is then. `#buffer refresh` after every line doesn't seem as slow as I feared, and I guess if the bumping really bothers me I'll set up buttons outside the buffer area.

**Comment by scandum** [comment]
It won't redraw after every line, but after every block of text, so it won't be too bad with a fast terminal. It should also be possible to shorten the input line with `#screen inputregion` to create some additional screen space. I think with some creativity, especially by utilizing MSLP, quite a bit can be done to make mobile mudding more convenient.
